### PR TITLE
Added documentation to the WHIP unit tests 

### LINKED ISSUE #66: Documentation: Add/update/improve documentation for the unit tests
The unit tests are currently barely documented. It would be helpful to other developers working on them to improve this documentation.

Documentation errors currently thrown for the unit tests when running PHPCS without excluding these issues:
```
FILE: tests\ConfigurationTest.php
------------------------------------------------------------------------------------------
FOUND 4 ERRORS AFFECTING 4 LINES
------------------------------------------------------------------------------------------
 12 | ERROR | Missing short description in doc comment
 19 | ERROR | Missing doc comment for function
    |       | testItReturnsANegativeNumberIfRequirementCannotBeFound()
 33 | ERROR | Missing doc comment for function testItReturnsAnEntryIfRequirementIsFound()
 47 | ERROR | Missing doc comment for function testIfRequirementIsConfigured()
------------------------------------------------------------------------------------------


FILE: tests\doubles\Whip_DismissStorageMock.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 13 | ERROR | Missing short description in doc comment
----------------------------------------------------------------------


FILE: tests\doubles\WPCoreFunctionsMock.php
----------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
----------------------------------------------------------------------
 12 | ERROR | Missing doc comment for function __()
 16 | ERROR | Missing doc comment for function esc_url()
----------------------------------------------------------------------


FILE: tests\MessageDismisserTest.php
----------------------------------------------------------------------
FOUND 3 ERRORS AFFECTING 3 LINES
----------------------------------------------------------------------
 13 | ERROR | Missing doc comment for function setUpBeforeClass()
 20 | ERROR | Missing short description in doc comment
 36 | ERROR | Missing short description in doc comment
----------------------------------------------------------------------


FILE: tests\MessagesManagerTest.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 13 | ERROR | Missing doc comment for function testHasMessages()
----------------------------------------------------------------------


FILE: tests\MessageTest.php
----------------------------------------------------------------------
FOUND 3 ERRORS AFFECTING 3 LINES
----------------------------------------------------------------------
 13 | ERROR | Missing doc comment for function testMessageHasBody()
 19 | ERROR | Missing short description in doc comment
 27 | ERROR | Missing short description in doc comment
----------------------------------------------------------------------


FILE: tests\RequirementsCheckerTest.php
------------------------------------------------------------------------------------------
FOUND 11 ERRORS AFFECTING 11 LINES
------------------------------------------------------------------------------------------
  13 | ERROR | Missing doc comment for function setUpBeforeClass()
  19 | ERROR | Missing doc comment for function testItReceivesAUsableRequirementObject()
  27 | ERROR | Missing short description in doc comment
  48 | ERROR | Missing doc comment for function
     |       | testItThrowsAnTypeErrorWhenInvalidRequirementIsPassedInPHP5()
  66 | ERROR | Missing doc comment for function testItOnlyContainsUniqueComponents()
  80 | ERROR | Missing doc comment for function testIfRequirementExists()
  90 | ERROR | Missing doc comment for function testCheckIfRequirementIsFulfilled()
  99 | ERROR | Missing doc comment for function testCheckIfPHPRequirementIsNotFulfilled()
 116 | ERROR | Missing doc comment for function testCheckIfRequirementIsNotFulfilled()
 133 | ERROR | Missing doc comment for function
     |       | testCheckIfRequirementIsFulfilledWithSpecificComparison()
 142 | ERROR | Missing doc comment for function
     |       | testCheckIfRequirementIsNotFulfilledWithSpecificComparison()
------------------------------------------------------------------------------------------


FILE: tests\VersionRequirementTest.php
---------------------------------------------------------------------------------
FOUND 15 ERRORS AFFECTING 12 LINES
---------------------------------------------------------------------------------
  13 | ERROR | Missing doc comment for function testNameAndVersionAreNotEmpty()
  20 | ERROR | Missing short description in doc comment
  28 | ERROR | Missing short description in doc comment
  36 | ERROR | Missing short description in doc comment
  44 | ERROR | Missing short description in doc comment
  52 | ERROR | Missing short description in doc comment
  60 | ERROR | Missing short description in doc comment
  68 | ERROR | Missing short description in doc comment
  76 | ERROR | Missing doc comment for function testGettingComponentProperties()
  84 | ERROR | Missing short description in doc comment
  84 | ERROR | Doc comment for parameter "$expectation" missing
  84 | ERROR | Doc comment for parameter "$component" missing
  84 | ERROR | Doc comment for parameter "$compareString" missing
  95 | ERROR | Missing doc comment for function dataFromCompareString()
 108 | ERROR | Missing short description in doc comment
---------------------------------------------------------------------------------
```

### PR DESCRIPTION
**Pull request by Dieterrr**
## Summary

This PR can be summarized in the following changelog entry:

* N/A

## Relevant technical choices:

* Added documentation (mostly descriptions) to the unit tests. 
* Removed the exceptions for the lack of documentation for these files from the `.phpcs.xml.dist` file.

## Test instructions
This PR can be tested by following these steps:
* As the changes are just comments in the unit tests nothing should be changed. 
* All unit tests should still pass. 

Fixes #66

### CODE REVIEW

File: tests/RequirementsCheckerTest.php [hunk 1]
```diff
@@ -87,6 +104,9 @@ public function testIfRequirementExists() {
 		$this->assertFalse( $checker->requirementExistsForComponent( 'mongodb' ) );
 	}
 
+	/**
+	 * If the requirement is fulfilled, there should be no message.
```
Comment (andizer) [inline tests/RequirementsCheckerTest.php]: It tests if the...

File: tests/RequirementsCheckerTest.php [hunk 2]
```diff
@@ -10,12 +10,18 @@
  */
 class RequirementsCheckerTest extends PHPUnit_Framework_TestCase {
 
+	/**
+	 * Instantiate our test class
```
Comment (andizer) [inline tests/RequirementsCheckerTest.php]: It isn't instantiating a class. It requires the WP Core functions mock.

File: tests/MessagesManagerTest.php [hunk 3]
```diff
@@ -10,6 +10,9 @@
  */
 class MessagesManagerTest extends PHPUnit_Framework_TestCase {
 
+	/**
+	 * Creates a MessagesManager, tests if it returns false without message, true when given a message.
```
Comment (andizer) [inline tests/MessagesManagerTest.php]: The method name says it testing has messaga.

**Comment by afercia** [comment]
CR 👍

**Pull request merged**